File: articles/science-of-flourishing-measuring-well-being/go/multidimensional_flourishing_index.go
package main

import "fmt"

func Mean(values []float64) float64 {
	total := 0.0
	for _, value := range values {
		total += value
	}
	return total / float64(len(values))
}

func main() {
	lifeSatisfaction := 7.6
	positiveAffect := 7.4
	negativeAffect := 2.5
	eudaimonicValues := []float64{7.5, 7.4, 7.3}
	positiveRelations := 7.7
	accomplishment := 7.4
	healthIndex := 7.5
	contextualSupport := 7.5
	stressLoad := 2.8

	hedonic := lifeSatisfaction + positiveAffect - negativeAffect
	eudaimonic := Mean(eudaimonicValues)
	integrated := 0.25*hedonic + 0.25*eudaimonic + 0.15*positiveRelations + 0.15*accomplishment + 0.15*healthIndex + 0.15*contextualSupport - 0.15*stressLoad

	fmt.Printf("Hedonic index: %.3f\n", hedonic)
	fmt.Printf("Eudaimonic index: %.3f\n", eudaimonic)
	fmt.Printf("Integrated flourishing index: %.3f\n", integrated)
}

File: articles/science-of-flourishing-measuring-well-being/rust/multidimensional_flourishing_index.rs
fn mean(values: &[f64]) -> f64 {
    values.iter().sum::<f64>() / values.len() as f64
}

fn main() {
    let life_satisfaction = 7.6;
    let positive_affect = 7.4;
    let negative_affect = 2.5;
    let eudaimonic_values = [7.5, 7.4, 7.3];
    let positive_relations = 7.7;
    let accomplishment = 7.4;
    let health_index = 7.5;
    let contextual_support = 7.5;
    let stress_load = 2.8;

    let hedonic = life_satisfaction + positive_affect - negative_affect;
    let eudaimonic = mean(&eudaimonic_values);
    let integrated =
        0.25 * hedonic +
        0.25 * eudaimonic +
        0.15 * positive_relations +
        0.15 * accomplishment +
        0.15 * health_index +
        0.15 * contextual_support -
        0.15 * stress_load;

    println!("Hedonic index: {:.3}", hedonic);
    println!("Eudaimonic index: {:.3}", eudaimonic);
    println!("Integrated flourishing index: {:.3}", integrated);
}
